Add LDAP connector documentation (#1304)
wso2/docs-ei: en/micro-integrator/docs/references/connectors/ldap-connector/ldap-server-configuration.md

3. Configure network parameters as follows and click next.
![image](../../../assets/img/connectors/ldap_connector/creating_a_new_connection.png)
4. Provide authentication parameters as follows and click finish.
    * Bind DN or user parameter - **uid=admin,ou=system**
    * Bind password - **admin**
5. Right click on newly created connection and select **Open Connection**.
![image](../../../assets/img/connectors/ldap_connector/open_connection.png)

### init

```xml
<ldap.init>
    <providerUrl>{$ctx:providerUrl}</providerUrl>
    <securityPrincipal>{$ctx:securityPrincipal}</securityPrincipal>
    <securityCredentials>{$ctx:securityCredentials}</securityCredentials>
    <secureConnection>{$ctx:secureConnection}</secureConnection>
    <disableSSLCertificateChecking>{$ctx:disableSSLCertificateChecking}</disableSSLCertificateChecking>
</ldap.init>
```

**providerUrl** : The URL of the LDAP server.</br>
**securityPrincipal** : The Distinguished Name(DN) of the admin of the LDAP Server.</br>
**securityCredentials** : The password of the LDAP admin.</br>
**secureConnection** : The boolean value for the secure connection.</br>

> **disableSSLCertificateChecking** : The boolean value to check whether certificate enable or not.</br>

Reviewer: ```suggestion
**disableSSLCertificateChecking** : The boolean value to check whether the certificate is enabled or not.</br>
```

Author: Updated.

> You can follow the following steps to import your LDAP certificate into wso2esb client’s keystore as follows:

Reviewer: ```suggestion
You can follow the steps below to import your LDAP certificate into wso2esb client’s keystore as follows:
```

Author: Updated.

> 1. To encrypt the connections, we'll need to configure a certificate authority (https://www.digitalocean.com/community/tutorials/how-to-encrypt-openldap-connections-using-starttls) 

Reviewer: ```suggestion
1. To encrypt the connections, you need to configure a certificate authority (https://www.digitalocean.com/community/tutorials/how-to-encrypt-openldap-connections-using-starttls) 
```

Author: Updated

and use it to sign the keys for the LDAP server.

> 2. Use the ESB Management Console or the following command to import that certificate into the ESB client keystore. 

Reviewer: Let's avoid recommending the management console.

Author: Removed.

> ```keytool -importcert -file <certificate file> -keystore <EI>/repository/resources/security/client-truststore.jks -alias "LDAP"```

Reviewer: Separate this into multiple lines and tab it.

Author: Updated

3. Restart the server and deploy the LDAP configuration.

#### Ensuring secure data
For security purposes, you should store securityCredentials in the WSO2 secure vault and make reference to it by using 

> an alias instead of hard-coding the actual value in the configuration file. For more information, see Working with 

Reviewer: The link is missing.

Author: Added relevant link.

Passwords.

#### Re-using LDAP configurations

> You can save the LDAP configuration as a [local entry](https://docs.wso2.com/display/EI620/Working+with+Local+Registry+Entries) 

Reviewer: Can we use the latest (GitHub-based) docs for this link?

Author: Added.
